static/js/settings_account.js

```diff
@@ -98,9 +106,78 @@ exports.add_custom_profile_fields_to_settings = function () {
             field_choices: field_choices,
         });
         $("#account-settings .custom-profile-fields-form").append(html);
+
+        if (is_user_field) {
+            var pill_container = $('.custom_user_field[data-field-id="' + field.id + '"] .pill-container').expectOne();
+            var pills = input_pill.create({
+                container: pill_container,
+                create_item_from_text: user_pill.create_item_from_email,
+                get_text_from_item: user_pill.get_email_from_item,
+            });
+            function append_user(user) {
+                if (user) {
+                    user_pill.append_person({
+                        pill_widget: pills,
+                        person: user,
+                    });
+                } else {
+                    blueslip.warn('Undefined user in function append_user');
+                }
+            }
+
+            function update_custom_user_field() {
+                var fields = [];
+                var user_id = user_pill.get_user_ids(pills);
+                if (user_id.length !== 1) {
+                    ui_report.message(i18n.t("Only one user allowed"), $("#custom-field-status"), 'alert-error');
+                    return;
+                }
+                fields.push({id: field.id, value: user_id[0]});
+                update_user_custom_profile_fields(fields);
+            }
+
+            if (value) {
+                var user = people.get_person_from_user_id(value);
+                append_user(user);
+            }
+            var input = pill_container.children('.input');
+            (function set_up_typeahead() {
+                input.typeahead({
+                    items: 5,
+                    fixed: true,
+                    dropup: true,
+                    source: function () {
+                        return user_pill.typeahead_source(pills);
+                    },
+                    highlighter: function (item) {
+                        return typeahead_helper.render_person(item);
+                    },
+                    matcher: function (item) {
+                        var query = this.query.toLowerCase();
+                        return (item.email.toLowerCase().indexOf(query) !== -1
+                                || item.full_name.toLowerCase().indexOf(query) !== -1);
+                    },
+                    sorter: function (matches) {
+                        return typeahead_helper.sort_recipientbox_typeahead(
+                            this.query, matches, "");
+                    },
+                    updater: function (user) {
+                        append_user(user);
+                        input.focus();
+                        update_custom_user_field();
+                    },
+                    stopAdvance: true,
+                });
+            }());
+
+            pills.onPillRemove(function () {
+                update_custom_user_field();
+            });
+        }
```

Comment:
We can reduce this code, as most of this is also used in `settings-user-group.js` for user group org setting. But I am confused in which file we should add this generic user pill functions.